legal-api/report-templates/certificateOfAmalgamation.html

```diff
@@ -34,19 +34,28 @@
                         <tr class="no-page-break">
                             <td class="col-50">
                                 <div class="{{'upper-text' if entity.legalName!='Not Available'}}">
-                                    and {{ entity.name }} ({{ entity.identifier }})
+                                    {% if entity.legalName %}
+                                        and {{ entity.legalName }} ({{ entity.identifier }})
+                                    {%else%}
+                                        and {{ entity.name }} ({{ entity.identifier }})
+                                    {%endif%}
                                 </div>
```

Comment:
Foreign companies will have 'legalName' instead of 'name' in the json (and possibly vice versa), this is to handle those situations.

Reply:
name field in amalgamatingBusinesses is not dependable as its not a [schema](https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/amalgamation_application.json#L38). if its present in filing json then it is purely for frontend purpose. Refer this loop https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/reports/business_document.py#L446, this is the right way to do this (but in PAID state business will be none so get ting with business identifier in amalgamatingBusinesses